client/src/main/java/io/pravega/client/segment/impl/NoSuchSegmentException.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.segment.impl;
+
+/**
+ * A segment does not exist and hence cannot be written to.
+ */
+public class NoSuchSegmentException extends IllegalStateException {
```

Comment:
I don't suppose there's a way we can unify this (and other client-side) exceptions with the ones used server-side?

Reply:
We could put them in shared, but I don't want to do that in this PR because it is a non-trivial refactoring. I have filed https://github.com/pravega/pravega/issues/1840